pymeasure/adapters/adapter.py

```diff
@@ -33,48 +37,135 @@ class Adapter:
 
     This class should only be inherited from.
 
-    :param preprocess_reply: optional callable used to preprocess strings
-        received from the instrument. The callable returns the processed string.
+    :param preprocess_reply: (deprecated) optional callable used to preprocess
+        strings received from the instrument. The callable returns the
+        processed string.
+    :param float query_delay: Time in s to wait after writing and before reading.
+    :param log: Parent logger of the 'Adapter' logger.
     :param kwargs: all other keyword arguments are ignored.
     """
 
-    def __init__(self, preprocess_reply=None, **kwargs):
+    def __init__(self, preprocess_reply=None, query_delay=0, log=None, **kwargs):
```

Comment:
Logging is a very useful feature. Thanks

Reply:
You're welcome. One reason for logging is, to write protocol tests observing the device communication: you set a value and the test generator generates the appropriate unit test.
I have a first, working version of this generator, which will be a follow up PR.